<launch>
  <arg name="capra_filters_ns" default="capra_filters" />

  <node pkg="nodelet" name="hide_robot" type="nodelet" args="load capra_filters/rectangle /$(arg capra_filters_ns)/nodelet_manager">
    <param name="output" value="no_robot" />
    <param name="height" value="61" />
    <param name="width" value="168" />
    <param name="input" value="/stereo/left/image_rect_color" />
    <param name="y" value="456" />
    <param name="x" value="295" />
    <param name="type" value="capra_filters/rectangle" />
  </node>

  <node pkg="nodelet" name="section_filter" type="nodelet" args="load capra_filters/section_filter /$(arg capra_filters_ns)/nodelet_manager">
    <param name="sections" value="7" />
    <param name="grass_max" value="224" />
    <param name="min_area" value="1000" />
    <param name="kernel_erode_width" value="2" />
    <param name="grass_min" value="163" />
    <param name="light_min" value="206" />
    <param name="kernel_dilate_width" value="1" />
    <param name="kernel_erode_height" value="2" />
    <param name="kernel_dilate_height" value="2" />

    <param name="input" value="no_robot" />
    <param name="output" value="section_filter" />
    <param name="type" value="section_filter" />
  </node>

  <node pkg="nodelet" name="hide_noise" type="nodelet" args="load capra_filters/rectangle /$(arg capra_filters_ns)/nodelet_manager">
    <param name="output" value="section_filter_filtered" />
    <param name="height" value="10" />
    <param name="width" value="700" />
    <param name="input" value="section_filter" />
    <param name="y" value="511" />
    <param name="x" value="0" />
    <param name="type" value="capra_filters/rectangle" />
  </node>


  <node pkg="nodelet" name="perspective_calibration" type="nodelet" args="load capra_filters/warp_perspective /$(arg capra_filters_ns)/nodelet_manager">
    <param name="a" value="0.5453037601275226" />
    <param name="b" value="1.1072528376015696" />
    <param name="c" value="-37.97825959884167" />
    <param name="d" value="0.21629236037945435" />
    <param name="e" value="2.6078006622487653" />
    <param name="f" value="7.717172745318196" />
    <param name="g" value="0.0010270904649316613" />
    <param name="h" value="0.01162129640408978" />
    <param name="i" value="1.0" />
    
    <param name="input" value="section_filter" />
    <param name="output" value="perspective_calibration" />
    <param name="type" value="capra_filters/warp_perspective" />
  </node>
  <node pkg="capra_filters" name="image_to_pointcloud" type="image_to_pointcloud.py">
    <param name="in" value="perspective_calibration" />
    <param name="out" value="/cloud_in" />
  </node>
</launch>
